rust/lance-index/src/vector/quantizer.rs

```diff
@@ -60,6 +50,22 @@ pub enum QuantizationType {
     Scalar,
 }
 
+impl TryFrom<&str> for QuantizationType {
+    type Error = Error;
+
+    fn try_from(value: &str) -> Result<Self> {
+        match value {
+            "FLAT" => Ok(Self::Flat),
+            "PQ" => Ok(Self::Product),
+            "SQ" => Ok(Self::Scalar),
+            _ => Err(Error::Index {
+                message: format!("Unknown quantization type: {}", value),
+                location: location!(),
+            }),
+        }
+    }
+}
```

Comment:
There is a `FromStr` trait you can implement instead of `TryFrom<&str>`.  If you implement `FromStr` then you can use [`str::parse`](https://doc.rust-lang.org/std/primitive.str.html#method.parse).

Reply:
Good point! Fixed